src/cmd-buildextend-gcp

```diff
@@ -113,18 +156,20 @@ def run_ore():
     checksum = sha256sum_file(build_tarball)
     size = os.path.getsize(build_tarball)
 
-    buildmeta['gcp'] = {'image': f"{base_name}-{args.build.replace('.', '-')}"}
-    buildmeta['images'].update({
-        'gcp': {
-            'path': gcp_tarball_name,
-            'sha256': checksum,
-            'size': size
-        }
-    })
-    write_json(buildmeta_path, buildmeta)
-    print(f"Updated: {buildmeta_path}")
-    shutil.rmtree(tmpdir)
+    # Update the meta to include our new output
+    build.meta['gcp'] = {
+        'image': f"{base_name}-{args.build.replace('.', '-')}",
```

Comment:
As I understand it, GCP is quite different from AWS in that a bootable image name is global.  It is however namespaced to a project, based on [this doc](https://cloud.google.com/compute/docs/images#community_supported_images).

So we probably want to include the project name here as well?

Reply:
A good reference here is to think about what the cluster/machine API needs.  The installer today injects the AMIs into the cluster API `machineset` objects.

I was looking at a [GCP example object](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/df71e50a06460a9379923fa454df181e1c5ef12e/cmd/clusterctl/examples/google/machines.yaml.template#L18) and it seems to just use `os` which is a public image.  It's not even clear to me that the GCP provider has support for starting an image...

Hum, maybe what the installer will need to do is copy the image from the project we own into the user's project.

(Have you tested booting an image that was uploaded using this code?)

Reply:
No, this is all still WIP :smile:. The majority of the code is the same (use of stat, name of the image, etc..). This PR attempts to push that code into using a `Build` class.

I'll hopefully context switch back to this later this week.

Reply:
/cc @darkmuggle

Reply:
Sharing of _registered_ compute images is not possible unless your images are "global."  Per [1], 

> It is not possible to grant roles to `allAuthenticatedUsers` or `allUsers` that allow access to images or snapshots. 

Sharing between organizations is allowed, but that seems to be an unsustainable path. What can be done, short of working with GCP, is to copy/upload the tarball then have the installer registered the object as a compute image. 

This leads me to conclude that including the  GCP section should have a bucket path and we should apply an ACL to allow copying.

[1]  https://cloud.google.com/compute/docs/images/sharing-images-across-projects

Reply:
It sounds like they're basically discouraging easy use of non-official images, which I totally understand.  Having the installer copy the image into the target project seems fine to me.  It's effectively the same as what we're doing with the AMIs in EC2 for a different reason (encryption).

From my PoV we don't need to have this bit finalized before merge but it'd be nice to have for sure (test case is uploading using one account and documenting the flow/steps to boot the image in another account with the data from the `meta.json`).